src/pytest_inmanta_lsm/remote_service_instance_async.py

```diff
@@ -186,6 +195,24 @@ async def history(self, *, since_version: int = 0) -> list[model.ServiceInstance
             filter={"version": f"ge:{since_version}"},
         )
 
+    async def diagnose(self, *, version: int) -> FullDiagnosis:
```

Comment:
Is there a specific reason why we need the `version` to be a kwarg?

Reply:
It is more a design choice.  I think it is clearer to force this argument name to be provided, as a simple number has very little meaning in a "diagnose" function.  While if you know it is the version, it makes a lot more sense.  It is also aligned with other helper methods on this class (where I made the same design choice).  Do you have an objection about it maybe?  I am not saying that I made the best design choice so I am open for discussion of course